x/stake/genesis.go

```diff
@@ -32,11 +32,16 @@ func InitGenesis(ctx sdk.Context, k Keeper, data GenesisState) {
 	k.setPool(ctx, data.Pool)
 	k.setNewParams(ctx, data.Params)
 	for _, validator := range data.Validators {
+		// Staking assumes bonded validators are already stored, need to force update
+		validator.PoolShares.Status = sdk.Unbonded
```

Comment:
we should never set the status directly - only update the status with validator.UpdateStatus(sdk.Unbonded)  - if we do a state genesis dump this comment isn't true - we can dump the state and have bonded validators

Reply:
In the latter case they should be set to bonded in `updateBondedValidatorsFull`? 

This is probably the wrong way to write this logic, though, we should perhaps update the store manually instead of calling `updateBondedValidatorsFull`?

Reply:
I think in genesis the validators should all already in the json be defaulted to unbonded (should have to set) and then we do a full update of the validator set using updateBondedValidatorsFull

if we are doing a state dump and reload, there may be unbonding validators etc. we should just properly dump their state, and properly reload the state and then call updateBondedValidatorsFull

In both cases we should just have the right information in the json and then run updateBondedValidatorsFull